### PR TITLE
Elasticsearch: Add developer documentation

### PR DESCRIPTION
**Pull request by ivanahuckova**

### CODE REVIEW

**Review by gtk-grafana — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/elasticsearch/docs/developer_documentation.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+## Development
+
+When developing for ElasticSearch, use `make devenv sources=elastic`. To specify a version, use `make devenv sources=elastic elastic_version=7.17.0`. In `devenv/docker/blocks/elastic/data/data.js`, you can update data to suit your debugging and testing needs. Additionally, ElasticSearch has a couple of debugging dashboards located in `devenv/dev-dashboards/datasource-ElasticSearch`.
+
+## Instrumentation
+
+The ElasticSearch data source has improved instrumentation with logs, metrics, traces, and dashboards. When debugging issues, it is useful to review the available telemetry signals.
+
+## Technical debt
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/elasticsearch/docs/developer_documentation.md:17]: Do we want to include this header without any content?

Comment (ivanahuckova) [reply public/app/plugins/datasource/elasticsearch/docs/developer_documentation.md:17]: I can add there sentence, plus I am wondering if we can remember more tech debt items

**Review by bohandley — APPROVED** [review]
Oh I see! To add the file which includes up to date Grafana docs. Nice!

File: public/app/plugins/datasource/elasticsearch/docs/developer_documentation.md (hunk 1 above)
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/docs/developer_documentation.md:17]:
```suggestion
## Technical debt
```
```suggestion
## Technical debt

Here is a list of our technical debt.
```

**Pull request merged**